packages/@lwc/integration-karma/test/light-dom/synthetic-shadow/index.spec.js

```diff
@@ -38,15 +38,8 @@ describe('Light DOM + Synthetic Shadow DOM', () => {
             expect(nodes.consumer.parentElement).toEqual(elm);
         });
 
-        // Issue [#2424]: Synthetic shadow returns an incorrect root node because the <p>
-        // doesn't have an owner key. However, synthetic shadow is no longer being changed.
-        // This test verifies that the existing behavior in synthetic shadow does not regress.
         it('getRootNode', () => {
-            const expectedRootNode = process.env.NATIVE_SHADOW
-                ? document // native, correct behavior
-                : nodes['consumer.shadowRoot']; // incorrect, existing behavior
-
-            expect(nodes.p.getRootNode()).toEqual(expectedRootNode);
+            expect(nodes.p.getRootNode()).toEqual(document);
```

Comment:
Does this have any other observable changes other than a top-level light DOM component slotting into a child synthetic shadow component, and then calling `getRootNode()` on the slotted element? What about manually-inserted light DOM nodes in the same scenario? What if the synthetic shadow component slots its own `<slot>` into its own child shadow component?

Reply:
This is the only scenario.
- Manually inserted light DOM nodes: In manually inserted DOM nodes (using `lwc:dom="manual"`), we navigate via `parentNodeGetter` to get the nearest shadow root.
- Synthetic shadow component slots its own `<slot>` into another child: it doesn't matter since what gets slotted still belongs to the root light DOM node (it has no shadow owner), so it's not a problem.